src/librustc/util/profiling.rs

```diff
@@ -103,11 +110,13 @@ macro_rules! define_categories {
 
                     json.push_str(&format!(
                         "{{ \"category\": \"{}\", \"time_ms\": {},\
-                            \"query_count\": {}, \"query_hits\": {} }},",
+                            \"query_count\": {}, \"query_hits\": {},\
+                            \"time_percent\": {} }},",
                         stringify!($name),
                         self.times.$name / 1_000_000,
                         total,
-                        format!("{:.2}", hit_percent)
+                        format!("{:.2}", hit_percent),
+                        format!("{:.2}", ((self.times.$name as f32) / total_time) * 100.0),
```

Comment:
I would avoid doing this. It adds no information for computers (i.e., the interpreters of the JSON), and does increase the amount of data provided -- and not in a lossless manner due to the f32 conversion.